kirin/piv/model_maker.py

```diff
@@ -51,6 +52,37 @@
 STOP_EVENT_DATETIME_MAP = {"arrivee": "arrival", "depart": "departure"}
 
 
+trip_piv_status_to_effect = {
+    "SUPPRESSION": TripEffect.NO_SERVICE,
+    "CREATION": TripEffect.ADDITIONAL_SERVICE,
+    "MODIFICATION_DETOURNEMENT": TripEffect.DETOUR,
+    "MODIFICATION_DESSERTE_SUPPRIMEE": TripEffect.REDUCED_SERVICE,
+    "MODIFICATION_LIMITATION": TripEffect.REDUCED_SERVICE,
+    "MODIFICATION_DESSERTE_AJOUTEE": TripEffect.MODIFIED_SERVICE,
+    "MODIFICATION_PROLONGATION": TripEffect.MODIFIED_SERVICE,
+    "RETARD": TripEffect.SIGNIFICANT_DELAYS,
+    "NORMAL": TripEffect.UNKNOWN_EFFECT,
+    "UNDEFINED": TripEffect.UNDEFINED,
+}
+
+trip_effect_order = {
+    TripEffect.ADDITIONAL_SERVICE: 0,
+    TripEffect.NO_SERVICE: 1,
+    TripEffect.DETOUR: 2,
+    TripEffect.REDUCED_SERVICE: 3,
+    TripEffect.MODIFIED_SERVICE: 4,
+    TripEffect.SIGNIFICANT_DELAYS: 5,
+    TripEffect.UNKNOWN_EFFECT: 6,
+    TripEffect.UNDEFINED: maxint,
+}
+
+MANAGED_EVENTS = ["RETARD", "SUPPRESSION"]
+
+
+def _get_trip_effect_order_from_piv_status(status):
+    return trip_effect_order[trip_piv_status_to_effect[status]]
```

Comment:
For safety, as we probably don't want to crash on those cases:
```suggestion
    return trip_effect_order.get(trip_piv_status_to_effect.get(status, TripEffect.UNDEFINED), maxint)
```